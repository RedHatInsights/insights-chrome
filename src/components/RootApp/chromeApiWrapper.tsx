import { ChromeAPI } from '@redhat-cloud-services/types';

const chromeApiWrapper = (chromeApi: ChromeAPI): ChromeAPI => {
  const internalApi = Object.keys(chromeApi).reduce((acc, curr) => {
    const key = curr as keyof ChromeAPI;
    const originalCall: ChromeAPI[typeof key] = chromeApi[key];
    if (typeof originalCall == 'function') {
      return {
        ...acc,
        [key]: (...args: Parameters<typeof originalCall>) => {
          console.error('Do not use chrome api call from window. It has been deprecated.');
          return originalCall(...args);
        },
      };
    }
    return acc;
  }, chromeApi);

  return internalApi;
};

export default chromeApiWrapper;
