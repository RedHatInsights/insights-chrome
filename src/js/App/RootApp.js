import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const RootApp = ({
    activeApp,
    activeLocation,
    appId,
    pageAction,
    pageObjectId
}) => {
    return (
        <Fragment>
            <div
                className="pf-c-drawer__content"
                data-ouia-page-type={activeApp || appId}
                data-ouia-page-group={activeLocation}
                data-ouia-safe='true'
                {...pageAction && { 'data-ouia-page-action': pageAction }}
                {...pageObjectId && { 'data-ouia-page-object-id': pageObjectId }}
            >
                <main className="pf-c-page__main pf-l-page__main" id="root" role="main">
                    <section
                        className="pf-m-light pf-c-page-header pf-c-page__main-section pf-m-light"
                        widget-type="InsightsPageHeader"
                    >
                        <div className="pf-c-content">
                            <h1
                                className="pf-c-title pf-m-2xl ins-l-page__header--loading"
                                widget-type='InsightsPageHeaderTitle'
                            >
                                <div className="ins-c-skeleton ins-c-skeleton__sm">
                                    &nbsp;
                                </div>
                            </h1>
                        </div>
                    </section>
                    <section className="pf-c-page__main-section pf-l-page__main-section--loading">
                        <div className="ins-c-spinner ins-m-center" role="status">
                            <span className="pf-u-screen-reader">Loading...</span>
                        </div>
                    </section>
                </main>
                <main className="pf-c-page__main" id="no-access"></main>
            </div>
            <aside className="pf-c-drawer__panel">
                <div className="pf-c-drawer__panel-body" />
            </aside>
        </Fragment>
    );
};

RootApp.propTypes = {
    appId: PropTypes.string,
    activeApp: PropTypes.string,
    activeLocation: PropTypes.string,
    pageAction: PropTypes.string,
    pageObjectId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

function stateToProps({ chrome: { activeApp, activeLocation, appId, pageAction, pageObjectId } }) {
    return ({ activeApp, activeLocation, appId, pageAction, pageObjectId });
}

export default connect(stateToProps, null)(RootApp);
