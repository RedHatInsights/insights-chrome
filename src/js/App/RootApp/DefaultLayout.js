import React, { memo, useEffect, useRef } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import GlobalFilter from '../GlobalFilter/GlobalFilter';
import { useScalprum } from '@scalprum/react-core';
import { Page, PageHeader, PageSidebar } from '@patternfly/react-core';
import { useLocation } from 'react-router-dom';
import { Header, HeaderTools } from '../Header/Header';
import Cookie from 'js-cookie';
import isEqual from 'lodash/isEqual';
import { onToggle } from '../../redux/actions';
import Routes from '../Routes';
import useOuiaTags from '../../utils/useOuiaTags';

import '../Sidenav/Navigation/Navigation.scss';
import './DefaultLayout.scss';
import { CROSS_ACCESS_ACCOUNT_NUMBER } from '../../consts';

const ShieldedRoot = memo(
  ({ hideNav, insightsContentRef, isGlobalFilterEnabled, initialized, Sidebar }) => {
    const dispatch = useDispatch();
    useEffect(() => {
      const navToggleElement = document.querySelector('button#nav-toggle');
      if (navToggleElement) {
        navToggleElement.onclick = () => dispatch(onToggle());
      }
    }, []);

    if (!initialized) {
      return null;
    }

    const selectedAccountNumber = Cookie.get(CROSS_ACCESS_ACCOUNT_NUMBER);
    const hasBanner = false; // Update this later when we use feature flags

    return (
      <Page
        isManagedSidebar={!hideNav}
        className={classnames({ 'ins-c-page__hasBanner': hasBanner, 'ins-c-page__account-banner': selectedAccountNumber })}
        header={<PageHeader logoComponent="div" logo={<Header />} showNavToggle={!hideNav} headerTools={<HeaderTools />} />}
        sidebar={hideNav ? undefined : <PageSidebar id="ins-c-sidebar" nav={Sidebar} />}
      >
        <div ref={insightsContentRef} className={classnames('ins-c-render', { 'ins-m-full--height': !isGlobalFilterEnabled })}>
          {isGlobalFilterEnabled && <GlobalFilter />}
          {selectedAccountNumber && <div className="ins-c-viewing-as">Viewing as Account {selectedAccountNumber}</div>}
          <Routes insightsContentRef={insightsContentRef} />
          <main className="pf-c-page__main" id="no-access"></main>
        </div>
      </Page>
    );
  },
  (prevProps, nextProps) => isEqual(prevProps, nextProps)
);

ShieldedRoot.propTypes = {
  hideNav: PropTypes.bool,
  insightsContentRef: PropTypes.object.isRequired,
  isGlobalFilterEnabled: PropTypes.bool.isRequired,
  initialized: PropTypes.bool,
  Sidebar: PropTypes.element,
};
ShieldedRoot.defaultProps = {
  useLandingNav: false,
  hideNav: false,
  isGlobalFilterEnabled: false,
  initialized: false,
};
ShieldedRoot.displayName = 'ShieldedRoot';

const RootApp = ({ globalFilterHidden, Sidebar }) => {
  const ouiaTags = useOuiaTags();
  const initialized = useScalprum(({ initialized }) => initialized);
  const { pathname } = useLocation();
  const hideNav = useSelector(({ chrome: { user } }) => !user || !Sidebar);
  const activeLocation = pathname.split('/')[1];
  /**
   * Using the chrome landing flag is not going to work because the appId is initialized inside the app.
   * We need the information before anything is rendered to determine if we use root module or render landing page.
   * This will be replaced once we can use react router for all pages. Landing page will have its own route.
   */
  const isLanding = pathname === '/';

  const isGlobalFilterEnabled =
    !isLanding && ((!globalFilterHidden && activeLocation === 'insights') || Boolean(localStorage.getItem('chrome:experimental:global-filter')));
  const insightsContentRef = useRef(null);

  return (
    <div id="chrome-app-render-root" className="pf-c-drawer__content" {...ouiaTags}>
      <ShieldedRoot
        isGlobalFilterEnabled={isGlobalFilterEnabled}
        hideNav={hideNav}
        insightsContentRef={insightsContentRef}
        initialized={initialized}
        Sidebar={Sidebar}
      />
    </div>
  );
};

RootApp.propTypes = {
  activeApp: PropTypes.string,
  pageAction: PropTypes.string,
  pageObjectId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  globalFilterHidden: PropTypes.bool,
  config: PropTypes.any,
  Sidebar: PropTypes.element,
};

export default RootApp;
