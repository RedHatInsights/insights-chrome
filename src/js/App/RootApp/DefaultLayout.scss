.ins-c-viewing-as {
  left: 0;
  top: 76px;
  width: 100%;
  padding: 2px 0 2px 20px;
  font-size: 10px;
  background-color: var(--pf-global--palette--blue-200);
  color: var(--pf-global--Color--100);
  opacity: 1;
  grid-area: banner;
}

.pf-c-page.ins-c-page__account-banner {
  grid-template-areas:
    "header header"
    "banner banner"
    "nav main";
  grid-template-rows: max-content max-content 1fr;
}

@media screen and (max-width: 1200px) {
  .pf-c-page.ins-c-page__account-banner {
    grid-template-areas:
      "header header"
      "main main";
  }
}
