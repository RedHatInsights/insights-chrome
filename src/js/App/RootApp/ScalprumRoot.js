import React, { lazy, Suspense, useState } from 'react';
import { ScalprumProvider } from '@scalprum/react-core';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { QuickStartContainer, useLocalStorage } from '@patternfly/quickstarts';

import DefaultLayout from './DefaultLayout';
import NavLoader from '../Sidenav/Navigation/Loader';
import { LazyQuickStartCatalog } from '../QuickStart/LazyQuickStartCatalog';
import { usePendoFeedback } from '../Feedback';
import { toggleFeedbackModal } from '../../redux/actions';

const Navigation = lazy(() => import('../Sidenav/Navigation'));
const LandingNav = lazy(() => import('../Sidenav/LandingNav'));

const loaderWrapper = (Component, props = {}) => (
  <Suspense fallback={<NavLoader />}>
    <Component {...props} />
  </Suspense>
);

const ScalprumRoot = ({ config, ...props }) => {
  const globalFilterRemoved = useSelector(({ globalFilter: { globalFilterRemoved } }) => globalFilterRemoved);
  const dispatch = useDispatch();
  const [activeQuickStartID, setActiveQuickStartID] = useLocalStorage('insights-quickstartId', '');
  const [allQuickStartStates, setAllQuickStartStates] = useLocalStorage('insights-quickstarts', {});
  const [quickStarts, setQuickStarts] = useState({});
  /**
   * Updates the available quick starts
   *
   * Usage example:
   * const { quickStarts } = useChrome();
   * quickStarts.set('applicationServices', quickStartsArray)
   *
   * @param {string} key App identifier
   * @param {array} qs Array of quick starts
   */
  const updateQuickStarts = (key, qs) => {
    const mergedQuickStarts = {
      ...quickStarts,
      [key]: qs,
    };
    setQuickStarts(mergedQuickStarts);
  };
  /**
   * Combines the quick start arrays
   * @returns Array of quick starts
   */
  const combinedQuickStarts = () => {
    const combined = [];
    for (const key in quickStarts) {
      combined.push(...quickStarts[key]);
    }
    return combined;
  };
  const quickStartProps = {
    quickStarts: combinedQuickStarts(),
    activeQuickStartID,
    allQuickStartStates,
    setActiveQuickStartID,
    setAllQuickStartStates,
    showCardFooters: false,
  };

  return (
    /**
     * Once all applications are migrated to chrome 2:
     * - define chrome API in chrome root after it mounts
     * - copy these functions to window
     * - add deprecation warning to the window functions
     */
    <QuickStartContainer {...quickStartProps}>
      <ScalprumProvider
        config={config}
        api={{
          chrome: {
            experimentalApi: true,
            ...window.insights.chrome,
            usePendoFeedback,
            toggleFeedbackModal: (...args) => dispatch(toggleFeedbackModal(...args)),
            quickStarts: {
              version: 1,
              set: updateQuickStarts,
              toggle: setActiveQuickStartID,
              Catalog: LazyQuickStartCatalog,
            },
          },
        }}
      >
        <Switch>
          <Route exact path="/">
            <DefaultLayout Sidebar={loaderWrapper(LandingNav)} {...props} globalFilterRemoved={globalFilterRemoved} />
          </Route>
          <Route path="/security">
            <DefaultLayout {...props} globalFilterRemoved={globalFilterRemoved} />
          </Route>
          <Route>
            <DefaultLayout Sidebar={loaderWrapper(Navigation)} {...props} globalFilterRemoved={globalFilterRemoved} />
          </Route>
        </Switch>
      </ScalprumProvider>
    </QuickStartContainer>
  );
};

ScalprumRoot.propTypes = {
  config: PropTypes.any,
};

export default ScalprumRoot;
