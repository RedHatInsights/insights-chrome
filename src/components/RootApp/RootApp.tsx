import React, { Suspense, lazy, memo, useContext, useEffect, useMemo } from 'react';
import { unstable_HistoryRouter as HistoryRouter, HistoryRouterProps } from 'react-router-dom';
import { HelpTopicContainer, QuickStart, QuickStartContainer, QuickStartContainerProps } from '@patternfly/quickstarts';
import { useAtomValue, useSetAtom } from 'jotai';
import chromeHistory from '../../utils/chromeHistory';
import { FeatureFlagsProvider } from '../FeatureFlags';
import ScalprumRoot from './ScalprumRoot';
import { LazyQuickStartCatalog } from '../QuickStart/LazyQuickStartCatalog';
import useQuickstartsStates from '../QuickStart/useQuickstartsStates';
import useHelpTopicState from '../QuickStart/useHelpTopicState';
import validateQuickstart from '../QuickStart/quickstartValidation';
import SegmentProvider from '../../analytics/SegmentProvider';
import { ITLess, chunkLoadErrorRefreshKey } from '../../utils/common';
import useUserSSOScopes from '../../hooks/useUserSSOScopes';
import { DeepRequired } from 'utility-types';
import ReactDOM from 'react-dom';
import ChromeAuthContext, { ChromeAuthContextValue } from '../../auth/ChromeAuthContext';
import { activeModuleAtom } from '../../state/atoms/activeModuleAtom';
import { scalprumConfigAtom } from '../../state/atoms/scalprumConfigAtom';
import { isDebuggerEnabledAtom } from '../../state/atoms/debuggerModalatom';
import { addQuickstartToAppAtom, clearQuickstartsAtom, populateQuickstartsAppAtom, quickstartsAtom } from '../../state/atoms/quickstartsAtom';

const NotEntitledModal = lazy(() => import('../NotEntitledModal'));
const Debugger = lazy(() => import('../Debugger'));

const RootApp = memo(({ accountId }: { accountId?: string }) => {
  const config = useAtomValue(scalprumConfigAtom);
  const { activateQuickstart, allQuickStartStates, setAllQuickStartStates, activeQuickStartID, setActiveQuickStartID } =
    useQuickstartsStates(accountId);
  const { helpTopics, addHelpTopics, disableTopics, enableTopics } = useHelpTopicState();
  const activeModule = useAtomValue(activeModuleAtom);
  const quickstartsData = useAtomValue(quickstartsAtom);
  const quickStarts = useMemo(() => Object.values(quickstartsData).flat(), [quickstartsData]);
  const clearQuickstarts = useSetAtom(clearQuickstartsAtom);
  const populateQuickstarts = useSetAtom(populateQuickstartsAppAtom);
  const addQuickstartToApp = useSetAtom(addQuickstartToAppAtom);

  useEffect(() => {
    clearQuickstarts(activeQuickStartID);
    if (activeModule) {
      let timeout: NodeJS.Timeout;
      const moduleStorageKey = `${chunkLoadErrorRefreshKey}-${activeModule}`;
      if (localStorage.getItem(moduleStorageKey) === 'true') {
        // The localStorage should either be true or null. A false value
        // can cause infinite loops. The timeout will remove the value after
        // ten seconds
        timeout = setTimeout(() => {
          localStorage.removeItem(moduleStorageKey);
        }, 10_000);
      }
      return () => {
        if (timeout) {
          clearTimeout(timeout);
        }
      };
    }
  }, [activeModule]);
  /**
   * Updates the available quick starts
   *
   * Usage example:
   * const { quickStarts } = useChrome();
   * quickStarts.set('applicationServices', quickStartsArray)
   *
   * @param {string} key App identifier
   * @param {array} qs Array of quick starts
   */
  const updateQuickStarts = (key: string, qs: QuickStart[]) => {
    populateQuickstarts({ app: key, quickstarts: qs });
  };

  const addQuickstart = (key: string, qs: QuickStart): boolean => {
    if (validateQuickstart(key, qs)) {
      addQuickstartToApp({ app: key, quickstart: qs });
      return true;
    }
    return false;
  };

  const quickStartProps: QuickStartContainerProps = {
    quickStarts,
    activeQuickStartID,
    allQuickStartStates,
    setActiveQuickStartID: setActiveQuickStartID as QuickStartContainerProps['setActiveQuickStartID'],
    setAllQuickStartStates: setAllQuickStartStates as unknown as QuickStartContainerProps['setAllQuickStartStates'],
    showCardFooters: false,
    language: 'en',
    alwaysShowTaskReview: true,
  };

  const helpTopicsAPI = {
    addHelpTopics,
    disableTopics,
    enableTopics,
  };

  const quickstartsAPI = {
    version: 1,
    set: updateQuickStarts,
    activateQuickstart,
    add: addQuickstart,
    toggle: setActiveQuickStartID,
    Catalog: LazyQuickStartCatalog,
    updateQuickStarts,
  };

  return (
    <HistoryRouter history={chromeHistory as unknown as HistoryRouterProps['history']}>
      <SegmentProvider>
        <FeatureFlagsProvider>
          {/* <CrossRequestNotifier /> */}
          <Suspense fallback={null}>
            <NotEntitledModal />
          </Suspense>
          <Suspense fallback={null}></Suspense>
          <QuickStartContainer {...quickStartProps}>
            <HelpTopicContainer helpTopics={helpTopics}>
              <ScalprumRoot config={config} quickstartsAPI={quickstartsAPI} helpTopicsAPI={helpTopicsAPI} />
            </HelpTopicContainer>
          </QuickStartContainer>
        </FeatureFlagsProvider>
      </SegmentProvider>
    </HistoryRouter>
  );
});

RootApp.displayName = 'MemoizedRootApp';

const AuthRoot = () => {
  const { user, login } = useContext(ChromeAuthContext) as DeepRequired<ChromeAuthContextValue>;
  const isDebuggerEnabled = useAtomValue(isDebuggerEnabledAtom);

  // verify use loged in scopes
  useUserSSOScopes(login);
  return (
    <>
      <Suspense fallback={null}>
        {user?.identity?.account_number && !ITLess() && isDebuggerEnabled && ReactDOM.createPortal(<Debugger user={user} />, document.body)}
      </Suspense>
      <RootApp accountId={user?.identity?.internal?.account_id} />
    </>
  );
};

export default AuthRoot;
