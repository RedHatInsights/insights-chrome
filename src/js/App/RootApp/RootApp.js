import React from 'react';
import { Router } from 'react-router-dom';
import { HelpTopicContainer, QuickStartContainer } from '@patternfly/quickstarts';

import { isBeta } from '../../utils';
import chromeHistory from '../../utils/chromeHistory';
import { FeatureFlagsProvider } from '../FeatureFlags';
import IDPChecker from './IDPChecker';
import ScalprumRoot from './ScalprumRoot';
import { useDispatch, useSelector } from 'react-redux';
import useQuickstartsStates from '../QuickStart/useQuickstartsStates';
import { populateQuickstartsCatalog } from '../../redux/actions';
import { LazyQuickStartCatalog } from '../QuickStart/LazyQuickStartCatalog';
import useHelpTopicState from '../QuickStart/useHelpTopicState';
import { SegmentProvider } from '../analytics/segment-analytics';

const RootApp = (props) => {
  const { allQuickStartStates, setAllQuickStartStates, activeQuickStartID, setActiveQuickStartID } = useQuickstartsStates();
  const { helpTopics, addHelpTopics, disableTopics, enableTopics } = useHelpTopicState();
  const dispatch = useDispatch();
  const activeModule = useSelector(({ chrome: { activeModule } }) => activeModule);
  const quickStarts = useSelector(
    ({
      chrome: {
        quickstarts: { quickstarts },
      },
    }) => Object.values(quickstarts).flat()
  );
  /**
   * Updates the available quick starts
   *
   * Usage example:
   * const { quickStarts } = useChrome();
   * quickStarts.set('applicationServices', quickStartsArray)
   *
   * @param {string} key App identifier
   * @param {array} qs Array of quick starts
   */
  const updateQuickStarts = (key, qs) => {
    dispatch(populateQuickstartsCatalog(key, qs));
  };

  const quickStartProps = {
    quickStarts,
    activeQuickStartID,
    allQuickStartStates,
    setActiveQuickStartID,
    setAllQuickStartStates,
    showCardFooters: false,
    language: 'en',
    alwaysShowTaskReview: true,
  };

  const helpTopicsAPI = {
    addHelpTopics,
    disableTopics,
    enableTopics,
  };

  const quickstartsAPI = {
    version: 1,
    set: updateQuickStarts,
    toggle: setActiveQuickStartID,
    Catalog: LazyQuickStartCatalog,
  };
  return (
    <Router history={chromeHistory} basename={isBeta() ? '/beta' : '/'}>
      <SegmentProvider activeModule={activeModule}>
        <FeatureFlagsProvider>
          <IDPChecker>
            {/* <CrossRequestNotifier /> */}

            <QuickStartContainer {...quickStartProps}>
              <HelpTopicContainer helpTopics={helpTopics}>
                <ScalprumRoot {...props} helpTopics={helpTopics} quickstartsAPI={quickstartsAPI} helpTopicsAPI={helpTopicsAPI} />
              </HelpTopicContainer>
            </QuickStartContainer>
          </IDPChecker>
        </FeatureFlagsProvider>
      </SegmentProvider>
    </Router>
  );
};

export default RootApp;
