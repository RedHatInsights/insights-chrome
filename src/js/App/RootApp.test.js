import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import RootApp from './RootApp';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('RootApp', () => {
    let initialState;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            chrome: {
                activeApp: 'some-app',
                activeLocation: 'some-location',
                appId: 'app-id'
            }
        };
    });

    it('should render correctly - no data', () => {
        const store = mockStore({ chrome: {} });
        const wrapper = render(
            <Provider store={store}>
                <RootApp/>
            </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = render(
            <Provider store={store}>
                <RootApp />
            </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with pageAction', () => {
        const store = mockStore({
            chrome: {
                ...initialState.chrome,
                pageAction: 'some-action'
            }
        });
        const wrapper = render(
            <Provider store={store}>
                <RootApp />
            </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with pageObjectId', () => {
        const store = mockStore({
            chrome: {
                ...initialState.chrome,
                pageObjectId: 'some-object-id'
            }
        });
        const wrapper = render(
            <Provider store={store}>
                <RootApp />
            </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with pageObjectId and pageAction', () => {
        const store = mockStore({
            chrome: {
                ...initialState.chrome,
                pageAction: 'some-action',
                pageObjectId: 'some-object-id'
            }
        });
        const wrapper = render(
            <Provider store={store}>
                <RootApp />
            </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
