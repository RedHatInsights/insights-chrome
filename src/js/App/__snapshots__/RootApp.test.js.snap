// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootApp should render correctly - no data 1`] = `
Array [
  <div
    class="pf-c-drawer__content"
  >
    <main
      class="pf-c-page__main pf-l-page__main"
      id="root"
      role="main"
    >
      <section
        class="pf-m-light pf-c-page-header pf-c-page__main-section pf-m-light"
        widget-type="InsightsPageHeader"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class="pf-c-title pf-m-2xl ins-l-page__header--loading"
            widget-type="InsightsPageHeaderTitle"
          >
            <div
              class="ins-c-skeleton ins-c-skeleton__sm"
            >
               
            </div>
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section pf-l-page__main-section--loading"
      >
        <div
          class="ins-c-spinner ins-m-center"
          role="status"
        >
          <span
            class="pf-u-screen-reader"
          >
            Loading...
          </span>
        </div>
      </section>
    </main>
    <main
      class="pf-c-page__main"
      id="no-access"
    />
  </div>,
  <aside
    class="pf-c-drawer__panel"
  >
    <div
      class="pf-c-drawer__panel-body"
    />
  </aside>,
]
`;

exports[`RootApp should render correctly 1`] = `
Array [
  <div
    class="pf-c-drawer__content"
    ouia-page-group="some-location"
    ouia-page-type="some-app"
  >
    <main
      class="pf-c-page__main pf-l-page__main"
      id="root"
      role="main"
    >
      <section
        class="pf-m-light pf-c-page-header pf-c-page__main-section pf-m-light"
        widget-type="InsightsPageHeader"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class="pf-c-title pf-m-2xl ins-l-page__header--loading"
            widget-type="InsightsPageHeaderTitle"
          >
            <div
              class="ins-c-skeleton ins-c-skeleton__sm"
            >
               
            </div>
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section pf-l-page__main-section--loading"
      >
        <div
          class="ins-c-spinner ins-m-center"
          role="status"
        >
          <span
            class="pf-u-screen-reader"
          >
            Loading...
          </span>
        </div>
      </section>
    </main>
    <main
      class="pf-c-page__main"
      id="no-access"
    />
  </div>,
  <aside
    class="pf-c-drawer__panel"
  >
    <div
      class="pf-c-drawer__panel-body"
    />
  </aside>,
]
`;

exports[`RootApp should render correctly with pageAction 1`] = `
Array [
  <div
    class="pf-c-drawer__content"
    ouia-page-action="some-action"
    ouia-page-group="some-location"
    ouia-page-type="some-app"
  >
    <main
      class="pf-c-page__main pf-l-page__main"
      id="root"
      role="main"
    >
      <section
        class="pf-m-light pf-c-page-header pf-c-page__main-section pf-m-light"
        widget-type="InsightsPageHeader"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class="pf-c-title pf-m-2xl ins-l-page__header--loading"
            widget-type="InsightsPageHeaderTitle"
          >
            <div
              class="ins-c-skeleton ins-c-skeleton__sm"
            >
               
            </div>
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section pf-l-page__main-section--loading"
      >
        <div
          class="ins-c-spinner ins-m-center"
          role="status"
        >
          <span
            class="pf-u-screen-reader"
          >
            Loading...
          </span>
        </div>
      </section>
    </main>
    <main
      class="pf-c-page__main"
      id="no-access"
    />
  </div>,
  <aside
    class="pf-c-drawer__panel"
  >
    <div
      class="pf-c-drawer__panel-body"
    />
  </aside>,
]
`;

exports[`RootApp should render correctly with pageObjectId 1`] = `
Array [
  <div
    class="pf-c-drawer__content"
    ouia-page-group="some-location"
    ouia-page-object-id="some-object-id"
    ouia-page-type="some-app"
  >
    <main
      class="pf-c-page__main pf-l-page__main"
      id="root"
      role="main"
    >
      <section
        class="pf-m-light pf-c-page-header pf-c-page__main-section pf-m-light"
        widget-type="InsightsPageHeader"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class="pf-c-title pf-m-2xl ins-l-page__header--loading"
            widget-type="InsightsPageHeaderTitle"
          >
            <div
              class="ins-c-skeleton ins-c-skeleton__sm"
            >
               
            </div>
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section pf-l-page__main-section--loading"
      >
        <div
          class="ins-c-spinner ins-m-center"
          role="status"
        >
          <span
            class="pf-u-screen-reader"
          >
            Loading...
          </span>
        </div>
      </section>
    </main>
    <main
      class="pf-c-page__main"
      id="no-access"
    />
  </div>,
  <aside
    class="pf-c-drawer__panel"
  >
    <div
      class="pf-c-drawer__panel-body"
    />
  </aside>,
]
`;

exports[`RootApp should render correctly with pageObjectId and pageAction 1`] = `
Array [
  <div
    class="pf-c-drawer__content"
    ouia-page-action="some-action"
    ouia-page-group="some-location"
    ouia-page-object-id="some-object-id"
    ouia-page-type="some-app"
  >
    <main
      class="pf-c-page__main pf-l-page__main"
      id="root"
      role="main"
    >
      <section
        class="pf-m-light pf-c-page-header pf-c-page__main-section pf-m-light"
        widget-type="InsightsPageHeader"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class="pf-c-title pf-m-2xl ins-l-page__header--loading"
            widget-type="InsightsPageHeaderTitle"
          >
            <div
              class="ins-c-skeleton ins-c-skeleton__sm"
            >
               
            </div>
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section pf-l-page__main-section--loading"
      >
        <div
          class="ins-c-spinner ins-m-center"
          role="status"
        >
          <span
            class="pf-u-screen-reader"
          >
            Loading...
          </span>
        </div>
      </section>
    </main>
    <main
      class="pf-c-page__main"
      id="no-access"
    />
  </div>,
  <aside
    class="pf-c-drawer__panel"
  >
    <div
      class="pf-c-drawer__panel-body"
    />
  </aside>,
]
`;
